katsdpimageutils/test/test_primary_beam_correction.py

```diff
@@ -0,0 +1,111 @@
+"""Tests for :mod:`katsdpimageutils.primary_beam_correction`."""
+import shutil
+import os
+
+import numpy as np
+from astropy.io import fits
+from nose.tools import assert_equal
+
+from ..primary_beam_correction import _get_value_from_history, weighted_average, inverse_variance, standard_deviation, write_new_fits # noqa E501 
+
+
+class TestStandardDeviation():
+    def setup(self):
+        self.arr = np.array([2, 4, 6, 8, 10]).astype(float)
+
+    def test_nonans(self):
+        data = self.arr
+        med, sd = standard_deviation(data)
+        assert_equal([med, sd], [6.0, 2.9652])
+
+    def test_nans(self):
+        data = self.arr
+        data[3] = np.nan
+        med, sd = standard_deviation(data)
+        assert_equal([med, sd], [5.0, 2.9652])
+
+    def test_allnan(self):
```

Comment:
I'm not sure you really need this test, because the raising (or not) of the Runtime warning doesn't really influence the outcome of the function.

Reply:
Noted.